### ISSUE TITLE
Cross-Origin reporting

### ISSUE THREAD

**Issue by toddreifsteck** [w3c/navigation-error-logging#17]
Cross-Origin reporting ensures that large scale DNS/TCP issues can be identified.

It also has a privacy/security concern where Parental Controls blocking site foo.com could then notify foo.com's reporting site that the site is blocked.

How significant would restriction of reporting to same origin be as a restriction?

**Comment by igrigorik** [comment]
From first-hand experience, same origin restriction would be a deal breaker for Google properties, all of which specify _multiple_ different report-uri origins to ensure delivery of error reports (via [Domain Reliability](https://docs.google.com/a/chromium.org/document/d/14U0YA4dlzNYciq2ke0StEMjomdBUN6ocSt1kN03HJ0s/edit?pli=1#)). I'd expect the same requirement for most other NEL users. In fact, I think we should _strongly encourage_ use of different origin for collecting NEL reports.

Also, note that imposing this restriction would break the real-time notification use-case: the UA would have to buffer the error report and wait until issue is resolved before it can deliver the report... and that's not guaranteed to happen (e.g. blocked access, attack against the user, network misconfiguration). See related discussion (in particular, istlsfastyet.com + TMo example) here: https://lists.w3.org/Archives/Public/public-web-perf/2015Jan/0025.html

---

I think the real issue here is distinguishing "opt-in" vs. malevolent failures... e.g. I may opt-in to block a particular site via parental controls or through malware protection (in which case NEL report shouldn't be sent), but I would like to distinguish that from cases where this is done without my consent (in which case NEL report should be sent). A couple of cases:

* if the blocking is done through UA logic (e.g. via UA configured blocking mechanism), then this is simple enough to handle.. and you shouldn't send the NEL report.
* if the blocking is done by some upstream component (e.g. firewall, proxy, etc), then it should return a meaningful error, not simply fail or timeout a connection... If the blocking is performed with appropriate opt-in, then it should advertise that to the user - e.g. return a 3XX redirect to a page describing why it was blocked.

In short, I think we just need to iterate on #25 and define what we mean by "navigation error", where the opt-in blocking cases are excluded?

**Comment by toddreifsteck** [comment]
Agreed
